src/Http/Handlers/ClientErrorHandler.php

```diff
@@ -15,7 +15,7 @@ public function __invoke(callable $handler)
         return function (RequestInterface $request, array $options = []) use ($handler) {
             return $handler($request, $options)->then(
                 function (ResponseInterface $response) use ($request) {
-                    if ($response->getStatusCode() >= 400) {
+                    if ($response->getStatusCode() >= 400 && $response->getStatusCode() !== 401) {
```

Comment:
It'd probably be a good idea to add a test case for this handler to ensure it ignores response code 401

Reply:
Great idea.